packages/jest-core/src/cli/index.ts

```diff
@@ -68,9 +71,22 @@ export async function runCLI(
     exit(0);
   }
 
+  let configsOfProjectsToRun = configs;
+  if (argv.selectProjects) {
+    const namesMissingWarning = getProjectNamesMissingWarning(configs);
+    if (namesMissingWarning) {
+      outputStream.write(namesMissingWarning);
```

Comment:
we usually go with `console.warn` for warnings, but I don't think it matters this much here

Reply:
`console.warn` will always write the string to `stdout`, whereas with outputStream we will write to either `stdout` or `stderr` depending on what is appropriate.

Here, if I replace `outputStream.write` by `console.warn` the tests will fail as they rely on the JSON output, and the JSON output can only be valid if all messages are written to `stderr` instead of `stdout`.